tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -6553,23 +6553,40 @@ Status ConvertCombinedNMS(OpConverterParams* params) {
 Status ConvertResize(OpConverterParams* params) {
   const auto& inputs = params->inputs;
   const auto& node_def = params->node_def;
-  TF_RETURN_IF_ERROR(
-      CheckInputsWeights(*params, {{"input", false}, {"size", true}}));
+  TF_RETURN_IF_ERROR(CheckInputsWeights(*params,
+      {{"input", TrtInputArg::kTensor}, {"size", TrtInputArg::kBoth}}));
   TF_RETURN_IF_ERROR(AllowDataTypes(
       *params, {DataType::DT_FLOAT, DataType::DT_HALF, DataType::DT_INT32}));
 
-  // Get input tensor. Transpose it from NHWC to NCHW.
+  // Get input tensor.
   ITensorProxyPtr inputs_tensor = inputs.at(0).tensor();
-
   TFTRT_RETURN_ERROR_IF_NULLPTR(inputs_tensor, params->node_def.name());
-
-  // Get output size. It must constain two values i.e. [H_out, W_out]
-  TRT_ShapedWeights weights = inputs.at(1).weights();
-  if (weights.count() != 2) {
-    return errors::Unimplemented("Resize to shape=[] is not supported, at ",
-                                 node_def.name());
+  const auto inputs_dims = inputs_tensor->getDimensions();
+  const bool static_input_shape = params->use_implicit_batch ||
+                                      HasStaticShape(inputs_dims);
+
+  // Check output size. It must constain two values i.e. [H_out, W_out]
+  const bool const_output_size = inputs.at(1).is_weights();
+  if (const_output_size) {
+    // Output size is given as a constant.
+    if (inputs.at(1).weights().count() != 2) {
+      return errors::Unimplemented("Resize to shape=[] is not supported, at ",
+                                   node_def.name());
+    }
```

Comment:
This code seems to imply is that if the rank is not 2, it is 0 ("shape=[]" in the error message). The error message it NOT added by this PR, so this is not your fault. Can we change the error message to "Resize requires 2D values for the new size"?

Reply:
This error message was updated in a commit yesterday.